primitives/types/metadata_generator.go

```diff
@@ -0,0 +1,179 @@
+package types
+
+import (
+	"reflect"
+	"strings"
+
+	sc "github.com/LimeChain/goscale"
+	"github.com/LimeChain/gosemble/constants/metadata"
+)
+
+type MetadataGenerator interface {
+	assignNewMetadataId(name string) (id int)
+	BuildMetadataTypeRecursively(t reflect.Value) int
+	IdsMap() map[string]int
+	constructFunctionName(input string) string
+	CallsMetadata(moduleName string, moduleFunctions map[sc.U8]Call, params *sc.Sequence[MetadataTypeParameter]) (MetadataType, int)
+}
+
+type MetadataTypeGenerator struct {
```

Comment:
should this be located in utils package instead? "primitives" is a reference to types defined by the Polkadot network, while this is a Golang-specific tool for internal project usage

Reply:
Yes, problem is that if I make **MetadataTypeGenerator** as a util, I have to import the primitives package in order to use Metadata types. And then since I pass the generator as a reference to the runtime extrinsic, which is then passed to the Metadata() method of the modules as a reference, I will have to change the signature of the Metadata() method of the modules which is located in the primitives/types package which will import the utils again.